src/gateway/Gateway.sol

```diff
@@ -51,7 +51,15 @@ contract Gateway is Auth, IGateway {
         _;
     }
 
+    event Received(address indexed sender, uint256 amount);
+    
+    // TODO Do we want to have a method that can withdraw the funds from the contract?
```

Comment:
I think we should add a version of `recoverToken` like this:
```solidity
    function recoverTokens(address token, address to, uint256 amount) external auth {
        if (token === address(0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee ) {
                payable(to).call{value: amount ether}
        } else {
                SafeTransferLib.safeTransfer(token, to, amount);
        }
    }
```

The address coming from: https://eips.ethereum.org/EIPS/eip-7528